src/holofuel/pages/CreateOfferRequest/CreateOfferRequest.js

```diff
@@ -62,19 +63,32 @@ const modePrepositions = {
 export default function CreateOfferRequest ({ history: { push } }) {
   const [mode, setMode] = useState(OFFER_MODE)
 
+  const { data: { holofuelUser: whoami = {} } = {} } = useQuery(HolofuelUserQuery)
   const { data: { holofuelHistoryCounterparties: agents } = {} } = useQuery(HolofuelHistoryCounterpartiesQuery)
   const createOffer = useOfferMutation()
   const createRequest = useRequestMutation()
 
   const { newMessage } = useFlashMessageContext()
+  const [hasDisplayedFlashMessage, setHasDisplayedFlashMessage] = useState(false)
 
   const [counterpartyId, setCounterpartyId] = useState('')
   const [counterpartyNick, setCounterpartyNick] = useState('')
   const [isCounterpartyFound, setCounterpartyFound] = useState(false)
+  const [isFormValid, setIsFormValid] = useState(false)
 
   useEffect(() => {
     setCounterpartyNick(presentAgentId(counterpartyId))
-  }, [counterpartyId])
+    if (counterpartyId.length === AGENT_ID_LENGTH && counterpartyId === whoami.id) {
```

Comment:
the length check here is redundant. `counterpartyId === whoami.id` implies `counterpartyId.length === AGENT_ID_LENGTH`

Reply:
otherwise, good change 👍